### PR TITLE
Adding featureDevSessionContext back for more rigorous file ignore in…

### PR DESCRIPTION
**Pull request by ashishrp-aws**
… payload creation.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
Problem: Customers seeing cases of gitignore files getting into file payload.
Solution: Currently chnageListManager seems to be not totally reliable, so adding back featureDevSessionContext back to do more rigorous file ignore check.

## Checklist
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by laileni-aws — APPROVED** [review]

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt [hunk 1]
```diff
@@ -151,7 +155,9 @@ class CodeScanSessionConfig(
                     val current = stack.pop()
 
                     if (!current.isDirectory) {
-                        if (!changeListManager.isIgnoredFile(current) && !files.contains(current.path)
+                        if (!changeListManager.isIgnoredFile(current) &&
+                            runBlocking { !featureDevSessionContext.ignoreFile(current, this) } &&
+                            !files.contains(current.path)
```
Comment (rli) [inline plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt:160]: please confirm there is no regression for large project trees

Comment (rli) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt:160]: still unclear why we can't start converging implementations https://github.com/aws/aws-toolkit-jetbrains/pull/4471

Comment (ashishrp-aws) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt:161]: We have different implementations for iterating through files(we need changeListManager which is not setup in 4471), different file size limits. So not sure about merging both implementations. But if other teams are ok with using changeListManager along with existing ignore setup, then I can try merging implementations.

Comment (ashishrp-aws) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt:161]: There is some slow down in file collection time.

**Pull request merged**